packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,19 +37,94 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// A position that the [FloatingActionButton] can dock to in a [Scaffold].
+/// 
+/// [getOffset] 
+class FabPosition {
```

Comment:
I'd prefer it if getOffset() was just a method, ScaffoldState provided the current geometry of its parts.

```
abstract class FabPosition {
  const FabPosition(); // to enable const constructors in subcalsses

  Offset getOffset(ScaffoldState scaffold);

  static const FabPosition centerFloat = const _CenterFloat();

  //...
}

class _CenterFloat extends FabPosition {
  const _CenterFloat();
  @override
  Offset getOffset(ScaffoldState scaffold) {
    final TextDirection textDirection = Directionality.of(scaffold.context).textDirection;
    final scaffoldSize = scaffold.size;
    final fabSize = scaffold.fabSize;
    final snackBarSize = scaffold.snackBarSize;
    final bottomSheetSize = scaffold.bottomSheetSize;
    // Or maybe just provide a scaffold.contentPadding getter whose insets
    // account for the snackbar, bottomsheet, appbar, etc.

    // Offset computing code is as before...
  }
}
```